### PR TITLE
Add repositoryId overloads to methods on I(Observable)RepositoriesClient

### PR DESCRIPTION
**Pull request by dampir**
As part of my work on #1120 I've added new overloads on I(Observable)RepositoriesClient to get access by repository id.

- [x] **Add overloads to IRepositoriesClient.**

	  Just add overloads of existing methods that use repositoryId to work with repo.
- [x] **Add overloads to IObservableRepositoriesClient.**

	  Just add overloads of existing methods that use repositoryId to work with repo.
- [x] **Add unit tests.**

	  I've added new unit tests that use repositoryId to work with repo that is just a full copy of existing tests that use (owner, name) key.
	  Also I've found out that not all methods are covered by tests and added them for new and for old methods.
- [x] **Add integration tests.**

	  I've added new integration tests that use repositoryId to work with repo that is just a full copy of existing tests that use (owner, name) key.
	  Also I've found out that not all methods are covered by tests and added them for new and for old methods.

### CODE REVIEW

File: Octokit.Reactive/Clients/ObservableRepositoriesClient.cs [hunk 1]
```diff
@@ -415,9 +496,34 @@ public IObservable<RepositoryContributor> GetAllContributors(string owner, strin
             Ensure.ArgumentNotNullOrEmptyString(owner, "owner");
             Ensure.ArgumentNotNullOrEmptyString(name, "name");
 
-            return GetAllContributors(owner, name, false, ApiOptions.None);
+            return GetAllContributors(owner, name, includeAnonymous, ApiOptions.None);
```
Comment (ryangribble) [inline Octokit.Reactive/Clients/ObservableRepositoriesClient.cs]: Nice pickup :+1:

File: Octokit.Reactive/Clients/ObservableRepositoriesClient.cs [hunk 2]
```diff
@@ -429,7 +535,29 @@ public IObservable<RepositoryContributor> GetAllContributors(string owner, strin
             if (includeAnonymous)
                 parameters.Add("anon", "1");
 
-            return _connection.GetAndFlattenAllPages<RepositoryContributor>(endpoint, parameters);
+            return _connection.GetAndFlattenAllPages<RepositoryContributor>(endpoint, parameters, options);
```
Comment (ryangribble) [inline Octokit.Reactive/Clients/ObservableRepositoriesClient.cs]: oh dear is this another place we missed passnig the `ApiOptions` through, on the previous changes?  nice pickup

**Comment by ryangribble** [comment]
Great stuff @dampir I couldn't spot anything missed, I also pulled down your branch and ran all the integration tests succesfully.  You've also fixed up plenty of other things like missing integration tests for the existing owner/name methods and even fixed a couple of bugs with the old code too :+1:

**Comment by ryangribble** [comment]
LGTM

**Comment by dampir** [comment]
@ryangribble thank you for positive feedback :+1:

File: Octokit.Reactive/Clients/ObservableRepositoriesClient.cs (hunk 2 above)
Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableRepositoriesClient.cs]: whoops, yep, great pickup @dampir!

Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableRepositoriesClient.cs]: whoops - great pickup @dampir!

**Pull request merged**